tests/TestKernel.php

```diff
@@ -53,17 +55,52 @@ function (ContainerBuilder $container) {
                     'symfonycasts.messenger_monitor.storage.doctrine_connection'
                 )->setPublic(true);
 
+                $container->setAlias('test.messenger.bus.default', 'messenger.bus.default')->setPublic(true);
+                $container->setAlias('test.messenger.receiver_locator', 'messenger.receiver_locator')->setPublic(true);
+
+                $messageHandlerDefinition = new Definition(TestableMessageHandler::class);
+                $messageHandlerDefinition->addTag('messenger.message_handler');
+                $container->setDefinition('test.message_handler', $messageHandlerDefinition);
+
                 $container->setParameter('kernel.secret', 123);
                 $container->prependExtensionConfig(
                     'framework',
                     [
-                        'session' => ['enabled' => true],
+                        'session' => [
+                            'enabled' => true,
+                            'storage_id' => 'session.storage.mock_file'
+                        ],
                         'router' => [
                             'resource' => 'kernel::loadRoutes',
+                            'type' => 'service',
                             'enabled' => true,
                         ],
+                        'messenger' => [
+                            'failure_transport' => 'failed',
+                            'transports' => [
+                                'queue' => [
+                                    'dsn' => 'doctrine://default?queue_name=queue',
+                                    'retry_strategy' => ['max_retries' => 0]
+                                ],
+                                'failed' => 'doctrine://default?queue_name=failed',
+                            ],
+                            'routing' => [
+                                TestableMessage::class => 'queue'
+                            ]
+                        ],
+                        'test' => true,
                     ]
                 );
+
+                if (!$container->hasDefinition('kernel')) {
+                    $container->register('kernel', static::class)
+                        ->setSynthetic(true)
+                        ->setPublic(true);
+                }
+
+                $kernelDefinition = $container->getDefinition('kernel');
+                $kernelDefinition->addTag('routing.route_loader');
```

Comment:
Is this needed?

Reply:
ok, not needed with `MicroKernelTrait` :)